File: 141-linked-list-cycle/141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode *root = head, *temp = head;
        if(head==NULL)
            return false;
        root=head;
        temp=head;
        while((temp->next!=NULL)&&(temp->next->next!=NULL))
        {
            root=root->next;
            temp=temp->next->next;
            if(root==temp)
                return true;
        }
        return false;
    }
};

File: 141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode slow = head;
        ListNode fast = head;
        
        boolean isLoop = false;
        while(fast != null && fast.next != null && fast.next.next != null) {
            fast = fast.next.next;
            slow = slow.next;
            if (slow == fast) {
                isLoop = true;
                break;
            }
        }
        
        return isLoop;
    }
}